mindmeld/tokenizer.py

```diff
@@ -179,12 +187,24 @@ def multiple_replace(self, text, compiled):
         """
         # For each match, look-up corresponding value in dictionary
         try:
-            return compiled.sub(self._one_xlat, text)
+
+            # Checks if replacement can be found in pre-defined match object (non-custom).
+            # If no key in match object, go to custom tokenizer handling in Exception.
+            filtered = compiled.sub(self._one_xlat, text)
+
+            # If no key error and custom tokenizer was involved
+            # then the token has unwanted special characters. Remove them and return.
+            if self._custom:
+                return self.compiled.sub(self._one_xlat, text)
+
+            # Return filtered list if non-custom tokenizer.
+            return filtered
+
         except KeyError:
             # In case of custom/app-specific tokenizer configuration
             logger.info("Using custom tokenizer configuration.")
             re_str = compiled.findall(text)
-            return "".join([e[0] if len(e) > 1 else e for e in re_str])
+            return "".join([list(filter(None, e))[0] for e in re_str])
```

Comment:
I think a note on the list->filter->none trick here would be good.

Reply:
Out of curiosity, why don't we do this:
```suggestion
            return "".join(list(filter(None, [e for e in re_str])))
```